globus_cli/parsing/endpoint_plus_path.py

```diff
@@ -0,0 +1,63 @@
+import click
+
+
+class EndpointPlusPath(click.ParamType):
+    """
+    Custom type for "<endpoint_id>:<path>"
+    Supports path being required and path being optional.
+
+    Always produces a Tuple.
+    """
+    name = 'endpoint plus path'
+
+    def __init__(self, *args, **kwargs):
+        # path requirement defaults to True, but can be tweaked by a kwarg
+        self.path_required = kwargs.pop('path_required', True)
+
+        super(EndpointPlusPath, self).__init__(*args, **kwargs)
+
+    def get_metavar(self, param):
+        """
+        Default metavar for this instance of the type.
+        """
+        return self.metavar
+
+    @property
+    def metavar(self):
+        """
+        Metavar as a property, to satisfy the slight brokenness of
+        click.Argument
+        Tracked against Click as an issue:
+        https://github.com/pallets/click/issues/674
+        """
+        if self.path_required:
+            return "ENDPOINT_ID:PATH"
+        else:
+            return "ENDPOINT_ID[:PATH]"
+
+    def convert(self, value, param, ctx):
+        """
+        ParamType.convert() is the actual processing method that takes a
+        provided parameter and parses it.
+        """
+        # passthrough conditions: None or already processed
+        if value is None or isinstance(value, tuple):
+            return value
+
+        # split the value on the first colon, leave the rest intact
+        splitval = value.split(':', 1)
+        # first element is the endpoint_id
+        endpoint_id = click.UUID(splitval[0])
```

Comment:
Ugh. Of course. Should we fix this in the CLI (easy) or in the SDK (also easy, somewhat less direct)?
I suspect others will create `uuid.UUID`s and pass them in doing exactly this kind of validation, or even under the misapprehension that we require it.